include/lsst/afw/image/Image.h

```diff
@@ -208,7 +208,7 @@ class Image : public ImageBase<PixelT> {
      * private
      */
     Image& operator=(const Image& rhs);
-    Image& operator=(Image&& rhs);
+    Image& operator=(Image&& rhs) noexcept;
```

Comment:
I get that we might want this to be `noexcept`, but have you checked that the implementation can actually guarantees it?  If we can't default it, then I'd guess not - though that may be something we could change if we migrate away from `boost::gil`.

Reply:
This ties in to my comment above -- a lot of these move-constructors and move-assignments were added some time after we'd already migrated to C++11, as part of implementing the new [explicit default or delete rule](https://developer.lsst.io/cpp/style.html#b-classes-should-explicitly-declare-or-delete-compiler-generated-members). Out of perhaps an excessive preoccupation with backwards-compatibility, I had the new move constructors delegate to the copy-constructors, emulating the behavior when only custom copy-constructors were defined.

In the long run we probably do want to undo that and switch to a move-based implementation, but the current implementation is most definitely not `noexcept`.